### ISSUE TITLE
Test that computed variables are lazy

### ISSUE THREAD

**Issue by tomwhite** [pystatgen/sgkit#356]
This was done in #349 for allele counting, but should be extended for all methods.

**Comment by tomwhite** [comment]
It's not sufficient to just test that the output array is a Dask array (although that is a good start), we need to be sure no intermediate computation has occurred. We could do this (in a test) by using a Dask scheduler that throws an error if a computation is attempted.

Something like

```python
import dask
import dask.array as da

def error_scheduler(*args, **kwargs):
    raise RuntimeError("Call to compute attempted")

with dask.config.set(scheduler=error_scheduler):
    x = da.random.uniform(low=0, high=10, size=(1000, 1000), chunks=(100, 100))
    y = x + x.T - x.mean(axis=0)

    print(y) # OK
    print(y.compute()) # triggers error
```

**Comment by jeromekelleher** [comment]
Good idea. Can we make some generic pytest infrastructure (like a decorator or something) where we do this automatically? Seems like this is something we'll want to test for every function, so having making it easy to do will be important.
